docs/NOW-spec.md

```diff
@@ -538,134 +271,79 @@ The NOW_VARBUF structure is used to represent variable-length buffers.
 
-**msgType (1 byte)**: The message type, specific to the message class.
+**msgFlags (2 bytes)**: Message flags.
 
-**msgFlags (2 bytes)**: The message flags, specific to the message type and class.
+| Flag | Meaning |
+|-------|---------|
+| NOW_NEGOTIATION_SET_HEATBEAT<br>0x0001 | `heartbeat` field is present. |
 
-#### NOW_STATUS
-A status code, with a structure similar to HRESULT.
+**versionMajor (1 byte)**: Major protocol version. Breaking changes in protocol should
+increment major version; Protocol implementations with different major version are not compatible.
 
-<table class="byte-layout">
-    <thead>
-        <tr>
-            <th>0</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th><th>6</th><th>7</th>
-            <th>8</th><th>9</th><th>10</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th>
-            <th>6</th><th>7</th><th>8</th><th>9</th><th>20</th><th>1</th><th>2</th><th>3</th>
-            <th>4</th><th>5</th><th>6</th><th>7</th><th>8</th><th>9</th><th>30</th><th>1</th>
-        </tr>
-    </thead>
-    <tbody>
-        <tr>
-            <td colspan="2">S</td>
-            <td colspan="1">Y</td>
-            <td colspan="1">Z</td>
-            <td colspan="4">class</td>
-            <td colspan="8">type</td>
-            <td colspan="16">code</td>
-        </tr>
-    </tbody>
-</table>
+**versionMinor (1 byte)**: Minor protocol version. Incremented when new non-breaking feature is added.
 
-**S (2 bits)**: Severity level.
+**systemCapset (4 bytes)**: System commands capabilities set.
 
-| Value | Meaning |
+| Flag | Meaning |
 |-------|---------|
-| NOW_SEVERITY_INFO<br>0 | Informative status |
-| NOW_SEVERITY_WARN<br>1 | Warning status |
-| NOW_SEVERITY_ERROR<br>2 | Error status (recoverable) |
-| NOW_SEVERITY_FATAL<br>3 | Error status (non-recoverable) |
+| NOW_CAP_SYSTEM_SHUTDOWN<br>0x0001 | System shutdown command support. |
 
-**Y (1 bit)**: Reserved. MUST be set to zero.
+**sessionCapset (4 bytes)**: Session commands capabilities set.
 
-**Z (1 bit)**: Reserved. MUST be set to zero.
+| Flag | Meaning |
+|-------|---------|
+| NOW_CAP_SESSION_LOCK<br>0x00000001 | Lock command support. |
+| NOW_CAP_SESSION_LOGOFF<br>0x00000002 | Logoff command support. |
+| NOW_CAP_SESSION_MSGBOX<br>0x00000004 | Message box command support. |
 
-**class (4 bits)**: Reserved. MUST be set to zero.
+**execCapset (4 bytes)**: Remote execution capabilities set.
 
-**type (8 bits)**: The status type.
+| Flag | Meaning |
+|-------|---------|
+| NOW_CAP_EXEC_STYLE_RUN<br>0x00000001 | Generic "Run" execution style. |
+| NOW_CAP_EXEC_STYLE_PROCESS<br>0x00000002 | CreateProcess() execution style. |
+| NOW_CAP_EXEC_STYLE_SHELL<br>0x00000004 | System shell (.sh) execution style. |
+| NOW_CAP_EXEC_STYLE_BATCH<br>0x00000008 | Windows batch file (.bat) execution style. |
+| NOW_CAP_EXEC_STYLE_WINPS<br>0x00000010 | Windows PowerShell (.ps1) execution style. |
+| NOW_CAP_EXEC_STYLE_PWSH<br>0x00000020 | PowerShell 7 (.ps1) execution style. |
```

Comment:
removed AppleScipt style variant until its implemented in next protocol version.